scripts/scr_start_load/scr_start_load.gml

```diff
@@ -1,547 +1,122 @@
-function scr_start_load(argument0, argument1, argument2) {
-    // argument0: the fleet object
-    // argument1: star object
-    // argument2: 1 for including escorts, 2 for no escorts
+function scr_start_load(fleet, load_from_star, escort_load) {
+    // fleet: the fleet object
+    // load_from_star: star object
+    // escort_load: 1 for including escorts, 2 for no escorts
 
     // this distributes the marines and vehicles to the correct ships if the chapter is fleet-based or a home-based chapter
-    /*
-    battle_barges
-    strike_cruisers
-    gladius
-    hunters
-    */
-
-    var splinter, _companies, _equip, remove_size, company_size, ship, ship_size, companies_loaded;
+	
+	
+	// i feel like there definatly is or should be a generic function for this????

```

Comment:
in an OOP language this would be trivial to do, but here we are with gml2.

the only thing I can think of right now, is map vehicles and marines into a common, simple data structure and do the ship loading operation on that simpler object. But this will probably require a big rewrite

Reply:
well i've know made a generic method set for moving marines within the marine struct i've been working on so that should solve this (sort of)...

Reply:
could you point me towards that method?

Reply:
made a comment in the correct place in marine_rpg pr